src/banded/BandedMatrix.jl

```diff
@@ -1007,3 +1007,59 @@ function resize(A::BandedSubBandedMatrix, n::Integer, m::Integer)
     l,u = bandwidths(A)
     _BandedMatrix(reshape(resize!(vec(copy(bandeddata(A))), (l+u+1)*m), l+u+1, m), n, l,u)
 end
+
+function sum(A::BandedMatrix; dims=:)
+    if(dims == :)
+        l, u = bandwidths(A)
+        ret = zero(eltype(A))
+        if(l + u < 0)
+            return ret
+        end
+        n, m = size(A)
+        #Only get nonempty bands
+        lower, upper = min(n-1, l), min(m-1, u)
+        for i = -lower:upper
+            ret += sum(A[band(i)])
+        end
+        ret
+    elseif(dims > 2)
+        A
+    elseif(dims == 2)
+        l, u = bandwidths(A)
+        n, m = size(A)
+        ret = zeros(eltype(A), (n, 1))
+        if(l + u < 0)
+            return ret
+        end
+        lower, upper = min(n-1, l), min(m-1, u)
+        for i = -lower:upper
+            b = A[band(i)]
```

Comment:
This is also allocating. I think it would be cleaner to just do a double forloop here